Sources/GRPC/Interceptor/ClientInterceptors.swift

```diff
@@ -97,3 +97,10 @@ open class ClientInterceptor<Request, Response> {
     context.cancel(promise: promise)
   }
 }
+
+#if compiler(>=5.6)
+// Interceptors sendability is unchecked: they have their own documented rules for thread safety
+// which we can't check. We require them to be sendable in order for interceptor factories to be
+// sendable and in turn, clients.
+extension ClientInterceptor: @unchecked Sendable {}
```

Comment:
Does this need to be `preconcurrency`, given that the class already existed and is `open` for subclassing?

Reply:
Ah, yes it sure does!

Reply:
I think it'd have to be both `@preconcurrency` _and_ `@unchecked`. But I think it's API breaking if we don't add the `@preconcurrency` annotation, because any adopters who are not concurrency aware but build in 5.6 mode will break.

Reply:
I think they inherit the `@unchecked` (none of the subclasses within `GRPCTests` complained about not being sendable, some of which have mutable state)?

Reply:
I don't think `@unchecked` should be inherited, so far as I know.

Reply:
However, I can't find any immediate citation for that belief.

Reply:
Hmmm, [this](https://forums.swift.org/t/improving-sendable/52377/21) seems relevant.

Reply:
As does the final post https://forums.swift.org/t/improving-sendable/52377/26:

> There are a number of unresolved issues with Sendable that could be tackled separately. Here's the list I know of:
> ...
> - Sendable with class types needs to be revisited. We likely want to require subclasses to restate Sendable conformance, decide if there are any cases where that conformance need not be @unchecked (e.g., all state is immutable and Sendable) and whether it can ever be inferred (same conditions as for removing @unchecked?).

I'm not sure exactly where that leaves us but I think `@preconcurrency` and `@unchecked` is probably the safest bet here.